libsolidity/codegen/ir/IRGenerationContext.cpp

```diff
@@ -137,20 +137,11 @@ void IRGenerationContext::initializeInternalDispatch(InternalDispatchMap _intern
 
 InternalDispatchMap IRGenerationContext::consumeInternalDispatchMap()
 {
-	m_directInternalFunctionCalls.clear();
-
 	InternalDispatchMap internalDispatch = move(m_internalDispatchMap);
 	m_internalDispatchMap.clear();
 	return internalDispatch;
 }
 
-void IRGenerationContext::internalFunctionCalledDirectly(Expression const& _expression)
-{
-	solAssert(m_directInternalFunctionCalls.count(&_expression) == 0, "");
-
-	m_directInternalFunctionCalls.insert(&_expression);
-}
-
 void IRGenerationContext::internalFunctionAccessed(Expression const& _expression, FunctionDefinition const& _function)
```

Comment:
Could remove the `_expression` parameter, but then we'd no longer have the assert below..

Reply:
Removing the assertion is fine. It was there to ensure that it's the right expression and not just some random expression unrelated to the function. If we don't need `_expression`, we don't need the assertion either.